FastCaloSimAnalyzer/FastCaloGpu/src/GeoLoadGpu_al.cxx

```diff
@@ -0,0 +1,364 @@
+  TestGeoKernel testGeo;
+  alpaka::exec<Acc>(queue
+		    , workdiv
+		    , testGeo
+		    , m_cells_d
+		    , m_regions_d
+		    , m_ncells
+		    , m_nregions
+		    , 14, 0, 32);
+  TestGeoKernel_G testGeo_G;
+  alpaka::exec<Acc>(queue
+		    , workdiv
+		    , testGeo_G
+		    , m_geo_d
+		    , 14, 0, 32);
+  alpaka::wait(queue);
+
+  std::cout << "TesGeo finished " << std::endl;
+
+  long long* c   = m_regions[14].cell_grid();
+  int        np  = m_regions[14].cell_grid_phi();
+  int        ne  = m_regions[14].cell_grid_eta();
+  int        idx = c[0 * np + 32];
+  Identifier Id  = m_cellid_array[idx];
+  std::cout << "From Host: Region[14]Grid[0][32]: index=" << idx << ", ID=" << Id
+            << ", HashCPU=" << ( *m_cells )[Id]->calo_hash() << ", neta=" << ne << ",  nphi=" << np
+            << ", eta=" << ( *m_cells )[Id]->eta() << std::endl;
+
+  return true;
+  // end test
+}
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+bool GeoLoadGpu::SanityCheck() {
+  // sanity check/test
+  QueueAcc queue(alpaka::getDevByIdx<Acc>(Idx{0}));
+
+  WorkDiv workdiv(Idx{1},Idx{1},Idx{1});
+  TestHelloKernel testHello;
+  alpaka::exec<Acc>(queue
+		    , workdiv
+		    , testHello);
+  TestCellKernel testCell;
+  alpaka::exec<Acc>(queue
+		    , workdiv
+		    , testCell
+		    , m_cells_d
+		    , 1872);
+  alpaka::wait(queue);
+
+  std::cout << " ID of 2000's cell " << m_cellid_array[2000] << std::endl;
+  Identifier Id = m_cellid_array[2000];
+  std::cout << "ID of cell 2000: " << ( *m_cells )[Id]->identify() << "hashid: " << ( *m_cells )[Id]->calo_hash()
+            << std::endl;
+  std::cout << "Size of Identify: " << sizeof( Identifier ) << "size of Region: " << sizeof( GeoRegion ) << std::endl;
+
+  std::cout << "GPU Kernel cell test lauched" << std::endl;
+
+  return true;
+}
+
+/* * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
+
+Rg_Sample_Index* GeoLoadGpu::get_sample_index_h_al()
+{
+  if(!pImpl) pImpl = new Impl();
+
+  pImpl->sample_index_buf_host = alpaka::allocBuf<Rg_Sample_Index, Idx>(alpaka::getDevByIdx<Host>(0u), static_cast<Idx>(m_max_sample));
+  return (Rg_Sample_Index*)alpaka::getPtrNative(pImpl->sample_index_buf_host);
+}
+
+GeoRegion* GeoLoadGpu::get_regions_al()
+{
+  if(!pImpl) pImpl = new Impl();
+
+  pImpl->regions_buf_host = alpaka::allocBuf<GeoRegion, Idx>(alpaka::getDevByIdx<Host>(0u), Idx{m_nregions});
+  return (GeoRegion*)alpaka::getPtrNative(pImpl->regions_buf_host);
+}
+
+long long* GeoLoadGpu::get_cell_grid_al(int neta, int nphi)
+{
+  if(!pImpl) pImpl = new Impl();
+
+  BufHostLongLong cells = alpaka::allocBuf<long long, Idx>(alpaka::getDevByIdx<Host>(0u), static_cast<Idx>(neta*nphi));
+  pImpl->reg_vec_host.push_back(cells);
+  return (long long*)alpaka::getPtrNative(cells);
+}
+
+bool GeoLoadGpu::LoadGpu_al() {
+  if ( !m_cells || m_ncells == 0 ) {
+    std::cout << "Geometry is empty " << std::endl;
+    return false;
+  }
+
```

Comment:
can we printout what backend device it's running on?

Reply:
We have a bunch of printouts in this function. Do you want to add device/backend info to all of them or only to this one?

Reply:
only this one - have a look at the other implementations and see if you can reproduce the syntax, eg:
`Executing on GPU: NVIDIA RTX A6000`

Reply:
OK, fixed. Also applied a couple of fixes to `CMakeLists.txt`